sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/implementation/converters/ChatThreadConverter.java

```diff
@@ -24,8 +23,9 @@ public static ChatThread convert(com.azure.communication.chat.implementation.mod
             .setTopic(obj.getTopic())
             .setCreatedOn(obj.getCreatedOn());
 
-        if (obj.getCreatedBy() != null && !obj.getCreatedBy().isEmpty()) {
-            chatThread.setCreatedBy(new CommunicationUserIdentifier(obj.getCreatedBy()));
+        if (obj.getCreatedByCommunicationIdentifier() != null) {
```

Comment:
Any reason to check for this one ? This is not an optional field, it should always be in the payload.

Reply:
If for some reason it does come back as empty from the API, not doing this check would result in a null pointer exception... if we do this check, then at least they can see the rest of the response and make a decision.

Reply:
why would this be different than any other field ? I don't think we should be validating the interface.

Reply:
we're not validating the interface here... but making the SDK more robust against unexcepted responses from the server... having null pointer exception thrown because of a blank createdBy coming back from the server would be very confusing and lead the developer down a wrong path. Populating the object according to what actually comes back from the server would help them understand what is going on.

Reply:
I am not sure that would make the sdks more robust. By that same token, every other mandatory field should be validated, There is no difference between createdOn, topic, ids, participants etc.. in almost all models if somehow the API is faulty.

Reply:
The difference is that if you pass a `null` to the constructor of the CommunicationIdentifier class, it will throw an exception.
This exception would be red-herring to the real issue.